boards/src/gps.cpp

```diff
@@ -1,84 +1,115 @@
 #include "gps.h"
 
-#include <cstdint>
+#include <HardwareSerial.h>
+#include <driver/uart.h>
 
-#include "TinyGPS++.h"
-#include "freertos/FreeRTOS.h"
-#include "freertos/semphr.h"
+#include "board_config.h"
+#include "storage.h"
 
 namespace gps {
 
-// Private
-namespace {
-constexpr uint16_t BAUD_RATE = 9600;
-constexpr uint8_t RX_PIN = 34;
-constexpr uint8_t TX_PIN = 12;
-constexpr uint8_t SERIAL_NUM = 1;
-constexpr uint16_t MAX_BYTES_PER_CYCLE = 256;
+constexpr inline uint8_t UART_BUS_NUMBER = UART_NUM_1;  // 0 is Serial
+constexpr inline TickType_t UART_TIMEOUT = 100;
+constexpr inline size_t UART_BUFFER_SIZE = 1024;
+constexpr inline size_t UART_QUEUE_SIZE = 20;
+constexpr inline size_t UART_PATTERN_QUEUE_SIZE = 20;
 
-Data gps_data;
+QueueHandle_t g_uart_queue;
```

Comment:
you are using many different queue implementation. Here `QueueHandle_t`, somewhere else is `SPSCQueue` and i think i saw yet another one different queue type